apps/emqx_lua_hook/src/emqx_lua_script.erl

```diff
@@ -169,8 +169,11 @@ on_client_subscribe(#{clientid := ClientId, username := Username}, _Properties,
     NewTopicFilters =
         lists:foldr(fun(TopicFilter, Acc) ->
                         case on_client_subscribe_single(ClientId, Username, TopicFilter, LuaState) of
-                            false -> Acc;
-                            NewTopicFilter -> [NewTopicFilter | Acc]
+                            false ->
```

Comment:
As long as `{stop, []}` is returned in line 180 instead of `stop`, the effect of canceling all subscriptions can be achieved. If the subscription option was updated and the topic name was changed before, as long as NewTopicFilters is not `[]`, it can be used normally updated subscription list.

Reply:
I tried only that at first (returning `{stop, []}` here) before changing `emqx_channel`.  That didn't work because the old `Replace` function in the end was returning the original topic filters.  :thinking:

Reply:
You're right. I didn't notice the code behind. 

So the problem is actually that originally we could only update subscriptions on the `client.subscribe` hook, but not reduce them.

I'm still not sure if we should add an extra `delete` field to the subscription options though. Because it looks like an implementation issue of `Replace` function.

Reply:
if it's purely internal, and the `delete` field does the job, why not.
probably not to complicate things, especially not the change the API contract in a bug fix release

maybe use a more search friendly macro for this field?

Reply:
Based on [this comment](https://github.com/emqx/emqx/pull/8288#discussion_r909612807), maybe we could rename it to something like `deny_subscription` then?